File: Math/2894. Divisible and Non-divisible Sums Difference/soln.py
class Solution:
    def sOD(self, n ,m):
        cnt = math.floor(n/m)
        return m*cnt*(cnt+1)/2
    def differenceOfSums(self, n: int, m: int) -> int:
        totalSum = n*(n+1)/2
        divSum = self.sOD(n,m)
        nonDiv = totalSum - divSum
        return (int)(nonDiv - divSum)

File: Math/2894. Divisible and Non-divisible Sums Difference/soln.cpp
class Solution {
public:
    int sumOfDivisible(int n, int m){
        int cnt = n/m;
        return m*cnt*(cnt+1)/2;
    }
    int differenceOfSums(int n, int m) {
        int totalSum = n*(n+1)/2;
        int divSum = sumOfDivisible(n,m);
        int nondivSum = totalSum - divSum;
        return nondivSum - divSum;
    }
};